### ISSUE TITLE
Call to undefined method Doctrine\ORM\Query\Expr::myCustomMethod()

### ISSUE THREAD

**Issue by esbobkov** [phpstan/phpstan-doctrine#62]
I have custom Expr class extended from Doctrine\ORM\Query\Expr

I think, PHPStan doesn't know about that class and trigger error 
```
In ExpressionBuilderDynamicReturnTypeExtension.php line 67:
                                                                                                         
  Attempted to call an undefined method named "myCustomMethod" of class "Doctrine\ORM\Query\Expr".
```

is it because `Doctrine\ORM\Query\Expr` harcoded in getClass() method?
How to fix that?

**Comment by ondrejmirtes** [comment]
Hi, how are you able to use custom Expr class? Do you extend your QueryBuilder? Can you show me your `$qb->expr()` method?

**Comment by esbobkov** [comment]
It's used as helper class for building custom DQL parts. 
instantiate only as `new CustomExpr()`. We able to generate some DQL parts and pass their to standard QueryBuilder methods.

So, our EntityManager doesn't know about `CustomExpr` too..)

Thinking about adding `exprClass` argument for `ExpressionBuilderDynamicReturnTypeExtension` costructor. What do You think?

**Comment by ondrejmirtes** [comment]
Can you create a small sample repo? Still can't imagine how it works and why PHPStan fails on it.

**Comment by esbobkov** [comment]
> Can you create a small sample repo? Still can't imagine how it works and why PHPStan fails on it.

I prepared an [example project](https://github.com/esbobkov/custom_expr_project)

**Comment by ondrejmirtes** [comment]
Hi, I just commited a fix to master that solves this: https://github.com/phpstan/phpstan-doctrine/commit/9fac3a7eb075a2594e0fb95aeeb96b6c0065e2d4

With a few caveats:

1) Right now, your code won't crash PHPStan, but it won't be checked for any errors. Run PHPStan with `doctrine.reportDynamicQueryBuilders` set to true (https://github.com/phpstan/phpstan-doctrine/blob/master/rules.neon#L3) to verify this. It might find other dynamic QBs that are impossible to verify.
2) To be able to verify code using your custom Expr class, unfortunately, the class has to contain a constructor, even if it's empty. This is currently a limitation of dynamic return type extension system in PHPStan: https://github.com/phpstan/phpstan/blob/master/src/Analyser/Scope.php#L1051-L1070

Please follow these steps and tell me if it solves your problem and I can tag a new version. Thanks.

**Comment by esbobkov** [comment]
This commit fixed my issue and thanks for caveats!

**Comment by ondrejmirtes** [comment]
Released as 0.11.4: https://github.com/phpstan/phpstan-doctrine/releases/tag/0.11.4

**Comment by ondrejmirtes** [comment]
You no longer need the empty constructor, thanks to 0.11.9: https://github.com/phpstan/phpstan/releases/tag/0.11.9 (and this commit: https://github.com/phpstan/phpstan/commit/5fc3a8d02358dacb7a63d0d18cb5155a5b93586d)
